validator/client/src/test/java/tech/pegasys/teku/validator/client/restapi/apis/PostRemoteKeysTest.java

```diff
@@ -0,0 +1,90 @@
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.validator.client.restapi.apis;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.Test;
+import tech.pegasys.teku.bls.BLSPublicKey;
+import tech.pegasys.teku.bls.BLSTestUtil;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiRequest;
+import tech.pegasys.teku.validator.client.ActiveKeyManager;
+import tech.pegasys.teku.validator.client.restapi.apis.schema.ExternalValidator;
+import tech.pegasys.teku.validator.client.restapi.apis.schema.PostKeyResult;
+import tech.pegasys.teku.validator.client.restapi.apis.schema.PostRemoteKeysRequest;
+
+public class PostRemoteKeysTest {
+  private final ActiveKeyManager keyManager = mock(ActiveKeyManager.class);
+  private final RestApiRequest request = mock(RestApiRequest.class);
+
+  @Test
+  void emptyRequest_shouldGiveEmptySuccess() throws JsonProcessingException {
+    final PostRemoteKeys endpoint = new PostRemoteKeys(keyManager);
+    final PostRemoteKeysRequest body = new PostRemoteKeysRequest();
+    when(request.getRequestBody()).thenReturn(body);
+
+    endpoint.handle(request);
+    verify(request).respondOk(List.of());
+  }
+
+  @Test
+  void validResponse_shouldGiveValidPostKeyResults()
+      throws JsonProcessingException, MalformedURLException {
+    final PostRemoteKeys endpoint = new PostRemoteKeys(keyManager);
+
+    List<ExternalValidator> externalValidators =
+        List.of(
+            new ExternalValidator(
+                BLSTestUtil.randomKeyPair(1).getPublicKey(),
+                Optional.of(new URL("http://host.com"))),
+            new ExternalValidator(BLSTestUtil.randomKeyPair(2).getPublicKey(), Optional.empty()));
+    final PostRemoteKeysRequest body = new PostRemoteKeysRequest(externalValidators);
+    when(request.getRequestBody()).thenReturn(body);
+
+    List<PostKeyResult> results = List.of(PostKeyResult.success(), PostKeyResult.success());
+    when(keyManager.importExternalValidators(externalValidators)).thenReturn(results);
+
+    endpoint.handle(request);
+    verify(request).respondOk(results);
+  }
+
+  @Test
+  void duplicate_shouldGiveDuplicateResponse()
+      throws JsonProcessingException, MalformedURLException {
+    final PostRemoteKeys endpoint = new PostRemoteKeys(keyManager);
+
+    BLSPublicKey publicKey = BLSTestUtil.randomKeyPair(1).getPublicKey();
+    URL url = new URL("http://host.com");
+
+    List<ExternalValidator> externalValidators =
+        List.of(
+            new ExternalValidator(publicKey, Optional.of(url)),
+            new ExternalValidator(publicKey, Optional.of(url)));
```

Comment:
another situation here we could look at is the same pubkey, different url.

because we're keyed off pubkey, it really should still be a duplicate... also an attempt to add an external pubkey thats already in use as a local source... that last one might be less easy to unit test...